distributed/comm/inproc.py

```diff
@@ -267,12 +275,7 @@ async def _listen(self):
             )
             # Notify connector
             conn_req.c_loop.add_callback(conn_req.conn_event.set)
-            try:
-                await self.on_connection(comm)
```

Comment:
test_adapt_quickly is getting stuck in:

https://github.com/dask/distributed/blob/b3f50ceff4012a64f8517da25060e24180c8a6d4/distributed/deploy/spec.py#L349

when SpecCluster tries to start 9 workers concurrently, with coverage enabled, after starting about 5 workers one of them times out in `connect(` closing the connection, and so causing the scheduler to stop listening